File: CS/SecuredExportExample.Module/SecurityObjects/ExportPermission.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.ExpressApp.Security;

namespace SecuredExportExample.Module.SecurityObjects {
    public class ExportPermission : IOperationPermission {
        public string Operation {
            get { return "Export"; }
        }
    }    
    public class ExportPermissionRequest : IPermissionRequest {
        public ExportPermissionRequest() {}
        public object GetHashObject() {
            return GetType().FullName;
        }
    }
    public class ExportPermissionRequestProcessor : PermissionRequestProcessorBase<ExportPermissionRequest> {
        private IPermissionDictionary permissions;
        public ExportPermissionRequestProcessor(IPermissionDictionary permissions) {
            this.permissions = permissions;
        }
        public override bool IsGranted(ExportPermissionRequest permissionRequest) {
            return (permissions.FindFirst<ExportPermission>() != null);
        }
    }
}

File: VB/SecuredExportExample.Module/SecurityObjects/ExportPermission.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.ExpressApp.Security

Namespace SecuredExportExample.Module.SecurityObjects
    Public Class ExportPermission
        Implements IOperationPermission

        Public ReadOnly Property Operation() As String Implements IOperationPermission.Operation
            Get
                Return "Export"
            End Get
        End Property
    End Class
    Public Class ExportPermissionRequest
        Implements IPermissionRequest

        Public Sub New()
        End Sub
        Public Function GetHashObject() As Object Implements IPermissionRequest.GetHashObject
            Return Me.GetType().FullName
        End Function
    End Class
    Public Class ExportPermissionRequestProcessor
        Inherits PermissionRequestProcessorBase(Of ExportPermissionRequest)

        Private permissions As IPermissionDictionary
        Public Sub New(ByVal permissions As IPermissionDictionary)
            Me.permissions = permissions
        End Sub
        Public Overrides Function IsGranted(ByVal permissionRequest As ExportPermissionRequest) As Boolean
            Return (permissions.FindFirst(Of ExportPermission)() IsNot Nothing)
        End Function
    End Class
End Namespace
